tayo-blog/src/Pages/Home.js

```diff
@@ -0,0 +1,47 @@
+import React, { useEffect, useState } from 'react';
+import data from '../data';
+import { Link } from 'react-router-dom';
+import axios from 'axios';
+
+export const Home = ({ setCurrentPost }) => {
+  const [blogs, setBlogs] = useState(data);
+
+  // useEffect(() => {
+  //   const blogPost = async () => {
+  //     const { data } = await axios.get('https://blogpostapi1.herokuapp.com/');
+  //     setBlogs(data.data);
+  //   };
+  //   blogPost();
+  // }, []);
+
+  const handleClick = (id) => {
+    setCurrentPost(id);
+  };
+
+  const allPost = blogs.map((blog) => {
+    const posts = blog.post.split(' ');
+    const splitedPost = posts.substr(0, 50);
```

Comment:
Why are you splitting `blog.post` first? You can use the `.substr` method on `blog.post` directly

Reply:
I actually do not want to show all my blog post at once i want to show the full post when you click on the read more

Reply:
`blog.post` is a `string`, you do not have to split the text first before you can get a sub-section from it. 
You can just `blog.post.substring(0,50)`